winpr/include/winpr/collections.h

```diff
@@ -124,7 +124,8 @@ extern "C"
 	WINPR_API void ArrayList_Clear(wArrayList* arrayList);
 	WINPR_API BOOL ArrayList_Contains(wArrayList* arrayList, const void* obj);
 
-	WINPR_API int ArrayList_Add(wArrayList* arrayList, void* obj);
+	WINPR_API WINPR_DEPRECATED(int ArrayList_Add(wArrayList* arrayList, const void* obj));
+	WINPR_API BOOL ArrayList_Append(wArrayList* arrayList, const void* obj);
```

Comment:
I don't see any advantage to have the obj as `const void *` as we wanna use the items in the list, so well they're not `const` at all.

Reply:
the advantage is that with the new API you can use `const` data without compiler warnings.

Reply:
The relevant function for manipulation is `ArrayList_GetItem` and that still returns a non `const` value.